sklearn/cross_validation.py

```diff
@@ -412,6 +412,16 @@ def __init__(self, labels, n_folds=3, shuffle=False, random_state=None):
                      " than the number of labels: {1}.").format(n_folds,
                                                                 n_labels))
 
+        if shuffle:
+            # In case of ties in label weights, label names are indirectly
+            # used to assign samples to folds. When shuffle=True, label names
+            # are randomized to obtain random fold assigments.
+            rng = check_random_state(self.random_state)
+            unique_labels = np.arange(n_labels, dtype=np.int)
+            rng.shuffle(unique_labels)
+            labels = unique_labels[labels]
+            unique_labels, labels = np.unique(labels, return_inverse=True)
```

Comment:
This line is useless: at this point `labels` is already an array comprised of values in `[0, n_labels - 1]` so there is no point to recompute the mapping.